test/pools/disjoint_pool_ctl.cpp

```diff
@@ -0,0 +1,153 @@
+// Copyright (C) 2025 Intel Corporation
+// Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exceptiongi
+
+#include <gtest/gtest.h>
+#include <umf/memory_pool.h>
+#include <umf/memory_provider.h>
+#include <umf/memory_provider_ops.h>
+#include <umf/pools/pool_disjoint.h>
+#include <umf/providers/provider_os_memory.h>
+
+#include "base.hpp"
+#include "utils_log.h"
+
+using umf_test::test;
+using namespace umf_test;
+
+#define RETURN_SUCCESS(ret) ASSERT_EQ(ret, UMF_RESULT_SUCCESS)
+
+// Encapsulating class for pool creation and destruction
+class PoolWrapper {
```

Comment:
why we can't use existing poolCreateExtUnique?

Reply:
The `poolFixtures.hpp` forces to compile with `*_compliance_test`:
```
/usr/bin/ld: CMakeFiles/test_ctl_api.dir/ctl/ctl_api.cpp.o: in function `umfPoolTest_malloc_compliance_Test::TestBody()':
/home/kfilipek/Development/unified-memory-framework/test/poolFixtures.hpp:485:(.text+0x8c00): undefined reference to `malloc_compliance_test(umf_memory_pool_t*)'
```
I can rewrite it as a separate PR and move poolCreateExtUnique as a separate header file.